### ISSUE TITLE
50 unseen classes of the CUB dataset?

### ISSUE THREAD

**Issue by ngthanhtin** [Glaciohound/CLORE#1]
Hello @Glaciohound ,
Thank you for sharing the code of your great work.
I read your paper and code but cannot find which of your 50 CUB classes were used for testing.
Can you share how you chose those classes, please?

And I also have a question on the CLIP_linear, you said that you add a linear layer on top of CLIP and fine-tune it. So what is the size of that linear layer (200 or 150)? if it is 150, then how can you use that to infer 50 test classes?

It would really help in my research.
Thank you so much.

**Comment by Glaciohound** [comment]
Hi Tin,

Thanks for your interests in our work! Let me respond to the questions one by one:

1. As describe in [this section](https://github.com/Glaciohound/CLORE/tree/main#cub-definitions) we follow the splitting of data classes of [TF-VAEGAN](https://github.com/akshitac8/tfvaegan). The split file is copied in our repo in `guidelines/CUB/att_splits.mat`. If one loads the file, it contains fields named `train_loc`, `val_loc`, `test_seen_loc`, and `test_unseen_loc` just as their names imply.

2. CLIP model is a feature matching model, which means it provides embeddings for each input (textual or visual) instead of directly emitting logits for specific classes. In our linear setting we keep this paradigm and the model structure untouched. So "linear layer" we fine-tuned is exactly the original feature output layer which has a output embedding of 512 as [explained here](https://huggingface.co/docs/transformers/model_doc/clip). For the test 50 classes the matching scores (or, logits) also come from dot-products between visual and textual features.

Hope this answers your questions and feel free to let us know if you have further questions.

Best,
Chi

**Comment by ngthanhtin** [comment]
Hi @Glaciohound , thank you for the clarification.
